packages/thirdweb/src/wallets/interfaces/wallet.ts

```diff
@@ -45,6 +46,7 @@ export type Wallet<TWalletId extends WalletId = WalletId> = {
 export type Account = {
   // REQUIRED
   address: Address;
+  type?: "local" | "json-rpc";
```

Comment:
hmmm, this basically aligns our `Account` type with viem's account type which I both like and dislike.

I'd prefer to keep the details of `account` internal to it instead of on the public api if possible.

Reply:
Agree keeping it internal is ideal, but an issue came up re: this type attribute when I was playing around with the latest release this weekend. When calling `createWalletClient`, we only pass the address to viem because our accounts don't line up with theirs 1:1, and when that function only gets an address it defaults to a JsonRpcAccount. 
 This causes issues in certain viem wallet actions that branch based on the account type. I specifically noticed it in `getAddresses` where it can't fetch the addresses from rpc because it's a local account.

Reply:
gotcha, this makes sense - for this useCase can we add a "hidden" flag on the account? Since the only use-case for this inside the adapter we should be able to then "check" for that hidden flag inside the adapter -> it solves the problem?

Reply:
basically what I'm thinking is we can add a `Symbol("accountType")` key on the account, which is completely internal so nobody will be able to access it (won't leak to public api) but inside the adapter code we can reference it by looking up for the same symbol

Reply:
let me know if that makes sense

Reply:
Yeah that makes sense. And should we just generate the other fields viem expects for an account (publicKey, source) within the adapter?

Reply:
actually publicKey might need to be another hidden field since privateKey doesn't stick around to generate it

Reply:
so maybe `privateKey` itself is the hidden field which can server both as the flag and the data required inside the adapter

Reply:
Are there any security concerns with that? Otherwise that makes sense

Reply:
I don't... think there are? Worth thinking through a couple of scenarios but if you have the pkey available *to put* there it means its already available in memory => if you're somehow attacking it you could already find it

Reply:
Yeah, was jw if third party libraries would be able to somehow access it and create downstream vulnerabilities unintentionally if they knew the symbol key. Tried and the only way I could get it was by converting the account to a string and pulling it from there, in which case the library would have to be either intentionally malicious or extremely determined to get to the raw private key to use it somehow.